### PR TITLE
Updated to Reduction Scheduler and Fix for Cast Op printing

### PR DESCRIPTION
**Pull request by kevinstephano**
You should look at the Cast Op change!

### CODE REVIEW

File: torch/csrc/jit/codegen/cuda/scheduler.cpp [hunk 1]
```diff
@@ -288,16 +288,17 @@ ReductionParams reductionHeuristic(
   } else {
     rparams.block_dim_x_ = red_outputs;
     rparams.block_dim_y_ = red_elems;
-    // TODO: Remove magic statement
-    rparams.block_dim_x_ /= 4;
   }
 
+  // std::cout << "Heur1: " << rparams.grid_dim_x_ << " " << rparams.grid_dim_y_
```
Comment (jjsjann123) [inline torch/csrc/jit/codegen/cuda/scheduler.cpp]: nitpicking: in case you missed this.

Comment (kevinstephano) [reply torch/csrc/jit/codegen/cuda/scheduler.cpp]: That's weird, I thought I removed that.

**Review by naoyam — APPROVED** [review]
LGTM, though I can't quite say much about scheduler.cpp.

File: torch/csrc/jit/codegen/cuda/scheduler.cpp (hunk 1 above)
Comment (kevinstephano) [reply torch/csrc/jit/codegen/cuda/scheduler.cpp]: Oh you are talking about the comment.

**Review by jjsjann123 — APPROVED** [review]
LGTM, saw some commented code in there that needs some cleaning.

**Pull request merged**